lib/net/parser.js

```diff
@@ -140,11 +140,6 @@ class Parser extends EventEmitter {
     let i = 0;
     for (; data[i + 4] !== 0 && i < 12; i++);
 
-    if (i === 12) {
```

Comment:
Looks like this was necessary because the length of `getcfheaders` is 12. I think this check relates with `bitcoind` serialization of the command, see serialization at [`src/protocol.h`](https://github.com/bitcoin/bitcoin/blob/fa0d0ff6e1bee60fde63724ae28a51aac5a94d4a/src/protocol.h#L28-L61) for `CMessageHeader`. [Protocol documentation](https://en.bitcoin.it/wiki/Protocol_documentation#Message_structure) states the command field size as 12, however, should be "...NULL padded (non-NULL padding results in packet rejected)". Is this something that needs to be updated in [BIP157](https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki#getcfheaders)?

Reply:
Possible that NULL padding doesn't apply since it's already max size.

Reply:
Yeah, true. Probably a question for when it's implemented in `bitcoind` which I haven't seen a branch that for yet when searching. Currently it reads as `char pchCommand[12]` and needs to be NULL terminated regardless, so there isn't the space.

Reply:
Note: protocol commits have been removed for consideration later.